CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(sail_riscv)
+
+# Enable CTest
+enable_testing()
+
+# Export compile_commands.json for IDE support.
+set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
+
+# Always use Position Independent Code. By default it is only used for
+# shared libraries (which require it), but you also need it for static
+# libraries if you link them into shared libraries.
+# Generally it just simplifies everything for a negligable performance cost.
+set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+
+# Don't allow undefined symbols. This is generally a pain.
+include(CheckLinkerFlag)
+
+check_linker_flag(C "-Wl,--no-undefined" LINKER_SUPPORTS_NO_UNDEFINED)
+if (LINKER_SUPPORTS_NO_UNDEFINED)
+    add_link_options("-Wl,--no-undefined")
+endif()
+
+# Extra CMake files.
+set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
+
+# These are the main requirements.
+# Don't use `REQUIRED` so that we can print custom help messages.
+find_package(ZLIB)
+if (NOT ZLIB_FOUND)
+    message(FATAL_ERROR "Zlib not found. Try 'sudo apt install zlib1g-dev' or 'sudo dnf install zlib-devel'.")
+endif()
+
+find_package(GMP)
+if (NOT GMP_FOUND)
+    message(FATAL_ERROR "GMP not found. Try 'sudo apt install libgmp3-dev' or 'sudo dnf install gmp-devel'.")
+endif()
+
+find_program(SAIL_BIN "sail")
+if (NOT SAIL_BIN)
+    message(FATAL_ERROR "Sail not found. See README.md for installation instructions.")
+endif()
+
+set(DEFAULT_ARCHITECTURES "rv32d;rv64d" CACHE STRING "Architectures to build by default (rv32f|rv64f|rv32d|rv64d)(_rvfi)? " )
+
+option(COVERAGE "Compile with Sail coverage collection enabled.")
+
+# Softfloat support.
+add_subdirectory("dependencies/softfloat")
+
+# Sail C runtime.
+add_subdirectory("sail_runtime")
+
+# Sail model generated C code.
+add_subdirectory("model")
+
+# Emulator binary.
+add_subdirectory("c_emulator")
+
+# Old pre-compiled riscv-tests.
+add_subdirectory("test/riscv-tests")
+
+# Convenience targets.
+add_custom_target(csim DEPENDS c_emulator_rv32d c_emulator_rv64d)
+add_custom_target(check DEPENDS generated_model_rv32d generated_model_rv64d)
+
+# TODO: Support static linking.
+# TODO: Add `interpret` target.
+# TODO: Add isabelle target.
+# TODO: Add lem target.
+# TODO: Add hol4 target.
+# TODO: Add latex target.
```

Comment:
Not sure we still care about the latex output now that all the docs are asciidoc?

Reply:
I assume ISAv10 will still need them to embed our own research extensions, and possibly the standard part too? Not sure if we've thought through how exactly ISAv10 will look in this world though.

Reply:
Yeah but that target would exist in sail-cheri-riscv and not here? Although maybe by ISAv10 it would just be a fork of this repo?

Reply:
Yeah very good question. We should probably discuss it on Thursday. I'll remove the todo here anyway.